tests/lib/c_lib/src/test_sqrt.c

```diff
@@ -128,35 +157,67 @@ double	error;
 					"max sqrtf error exceeded");
 		}
 	}
+	zassert_true(max_error < 0x80, "huge errors in sqrt implementation");
```

Comment:
I suspect we can tighten this bounds up considerably. Get some results from picolibc and newlib and use those to compute the max error to use here.

Reply:
For the test cases I have, the max error for PICOLIBC is 1, and the max error for LIBC_MINIMAL is 2. When I had coding errors there were errors in the range 0x00FFFFFF. I am just trying to filter out serious coding errors.

Reply:
Even minor coding errors are important here -- let's use the bounds of '2' so we can detect those.